onnxruntime/core/providers/cpu/reduction/reduction_ops.cc

```diff
@@ -688,18 +688,19 @@ FastReduceKind OptimizeShapeForFastReduce(gsl::span<const int64_t> input_shape,
   return FastReduceKind::kNone;
 }
 
-void ValidateCommonFastReduce(const Tensor* axes_tensor) {
-  ORT_ENFORCE(axes_tensor != nullptr, "Axes input is null");
-  ORT_ENFORCE(axes_tensor->Shape().NumDimensions() == 1,
-              "An axes tensor must be a vector tensor.");
-}
-
 // template <typename T, typename TVAL>
 bool CommonFastReduceCopy(OpKernelContext* ctx, TensorShapeVector& input_axes, bool noop_with_empty_axes) {
   if (ctx->InputCount() == 2) {
     // second input holds the axes.
+    // the argument is optional
     const Tensor* axes_tensor = ctx->Input<Tensor>(1);
-    ValidateCommonFastReduce(axes_tensor);
+    if (axes_tensor == nullptr) {
+      return false;
```

Comment:
This looks correct. But I wonder if it can be made more efficient. My interpretation of this case is that it is the same as an empty axes tensor, and can be optimized as below if `noop_with_empty_axes` is true. Does that sound right? If so, we should just make the following lines upto `input_axes.insert(...)` conditional on `axes_tensor != nullptr` and continue on?